xbmc/FileItemList.cpp

```diff
@@ -705,7 +705,31 @@ void CFileItemList::RemoveExtensions()
     m_items[i]->RemoveExtension();
 }
 
-void CFileItemList::Stack(bool stackFiles /* = true */)
+void CFileItemList::ConvertDiscFoldersToFiles()
+{
+  for (const auto& item : m_items)
+  {
+    const std::string playPath = VIDEO::UTILS::GetOpticalMediaPath(*item);
+    if (!playPath.empty())
+      ConvertDiscFolderToFile(item, playPath);
+  }
+}
+
+void CFileItemList::ConvertDiscFolderToFile(const std::shared_ptr<CFileItem>& item,
+                                            const std::string& playPath)
+{
+
+  item->SetPath(playPath); // updated path (for DVD/Bluray files)
+  item->m_bIsFolder = false;
+  item->SetLabel2("");
+  item->SetLabelPreformatted(true);
+  m_sortDescription.sortBy = SortByNone; // sorting is now broken
```

Comment:
Now broken? wtf? Reads not okay. I saw that you just copied this code from elsewhere, but anyway, should be fixed.